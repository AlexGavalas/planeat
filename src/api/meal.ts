import { type SupabaseClient } from '@supabase/supabase-js';

import { type EditedMeal } from '~types/meal';
import { type Database } from '~types/supabase';

type FetchMealsProps = {
    startDate: string;
    endDate: string;
    supabase: SupabaseClient<Database>;
    userId: number;
};

export const fetchMeals = async ({
    startDate,
    endDate,
    supabase,
    userId,
}: FetchMealsProps) => {
    const result = await supabase
        .from('meals')
        .select('*')
        .gte('day', startDate)
        .lte('day', endDate)
        .eq('user_id', userId);

    return result;
};

type DeleteMealProps = {
    supabase: SupabaseClient<Database>;
    userId: number;
    deletedIds: string[];
};

export const deleteMeals = async ({
    deletedIds,
    supabase,
    userId,
}: DeleteMealProps) => {
    const { error } = await supabase
        .from('meals')
        .delete()
        .in('id', deletedIds)
        .eq('user_id', userId);

    return { error };
};

type UpdateMealsProps = {
    supabase: SupabaseClient<Database>;
    editedMeals: EditedMeal[];
    userId: number;
};

export const updateMeals = async ({
    editedMeals,
    supabase,
    userId,
}: UpdateMealsProps) => {
    const { error } = await supabase
        .from('meals')
        .upsert(editedMeals)
        .eq('user_id', userId);

    return { error };
};

type CreateMealsProps = {
    supabase: SupabaseClient<Database>;
    newMeals: EditedMeal[];
    userId: number;
};

export const createMeals = async ({
    newMeals,
    supabase,
    userId,
}: CreateMealsProps) => {
    const newMealsWithUserId = newMeals.map((meal) => ({
        ...meal,
        user_id: userId,
    }));

    const { error } = await supabase.from('meals').insert(newMealsWithUserId);

    return { error };
};
