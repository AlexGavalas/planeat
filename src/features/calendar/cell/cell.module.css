.cell {
    height: 100%;
    display: grid;
    gap: 5px;
    border: 1px solid black;
    border-radius: 3px;
    padding: 5px;
    white-space: pre-wrap;
}
