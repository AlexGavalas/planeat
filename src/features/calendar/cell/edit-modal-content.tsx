import { Button, Group, Textarea } from '@mantine/core';
import { useModals } from '@mantine/modals';
import { useTranslation } from 'next-i18next';
import { type FormEventHandler, useState } from 'react';

interface ModalContentProps {
    deleteMeal: () => Promise<void>;
    handleSave: (meal: string) => Promise<void>;
    initialMeal: string;
}

export const ModalContent = ({
    handleSave,
    initialMeal,
    deleteMeal,
}: ModalContentProps) => {
    const { t } = useTranslation();

    const modals = useModals();
    const [error, setError] = useState('');

    const closeModal = () => modals.closeAll();

    const onSubmit: FormEventHandler<HTMLFormElement> = (e) => {
        e.preventDefault();

        const meal = new FormData(e.currentTarget).get('meal')?.toString();

        if (!meal) return setError(t('errors.meal_empty').toString());

        handleSave(meal);
        closeModal();
    };

    const onDelete = () => {
        deleteMeal();
        closeModal();
    };

    return (
        <form onSubmit={onSubmit}>
            <Group direction="column" spacing="sm" grow>
                <Textarea
                    data-autofocus
                    name="meal"
                    placeholder={t('meal_placeholder').toString()}
                    label={t('meal_label')}
                    defaultValue={initialMeal}
                    autosize
                    minRows={5}
                    maxRows={20}
                    error={error}
                    onFocus={() => setError('')}
                />
                <Group position="apart" spacing="sm">
                    <Button variant="light" color="red" onClick={closeModal}>
                        {t('cancel')}
                    </Button>
                    <Group>
                        <Button
                            color="red"
                            hidden={!initialMeal}
                            onClick={onDelete}
                        >
                            {t('delete')}
                        </Button>
                        <Button type="submit">{t('save')}</Button>
                    </Group>
                </Group>
            </Group>
        </form>
    );
};
