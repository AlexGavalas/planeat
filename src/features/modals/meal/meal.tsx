import { Button, Group, Stack, Textarea } from '@mantine/core';
import { useModals } from '@mantine/modals';
import { useTranslation } from 'next-i18next';
import { type FormEventHandler, type MouseEventHandler, useState } from 'react';

type MealModalProps = {
    deleteMeal: () => Promise<void> | void;
    handleSave: (meal: string) => Promise<void> | void;
    initialMeal: string;
};

export const MealModal = ({
    handleSave,
    initialMeal,
    deleteMeal,
}: MealModalProps) => {
    const { t } = useTranslation();
    const modals = useModals();
    const [error, setError] = useState('');

    const closeModal = () => {
        modals.closeAll();
    };

    // eslint-disable-next-line @typescript-eslint/no-misused-promises -- async event handler
    const onSubmit: FormEventHandler<HTMLFormElement> = async (e) => {
        e.preventDefault();

        const meal = new FormData(e.currentTarget).get('meal')?.toString();

        if (!meal) {
            setError(t('errors.meal_empty'));
            return;
        }

        await handleSave(meal);

        closeModal();
    };

    // eslint-disable-next-line @typescript-eslint/no-misused-promises
    const onDelete: MouseEventHandler<HTMLButtonElement> = async () => {
        await deleteMeal();
        closeModal();
    };

    return (
        <form onSubmit={onSubmit}>
            <Stack gap="sm">
                <Textarea
                    data-autofocus
                    name="meal"
                    placeholder={t('meal_placeholder')}
                    label={t('meal_label')}
                    defaultValue={initialMeal}
                    autosize
                    minRows={5}
                    maxRows={20}
                    error={error}
                    onFocus={() => {
                        setError('');
                    }}
                />
                <Group justify="space-between" gap="sm">
                    <Button variant="light" color="red" onClick={closeModal}>
                        {t('cancel')}
                    </Button>
                    <Group>
                        <Button
                            color="red"
                            hidden={!initialMeal}
                            onClick={onDelete}
                        >
                            {t('delete')}
                        </Button>
                        <Button type="submit">{t('save')}</Button>
                    </Group>
                </Group>
            </Stack>
        </form>
    );
};
