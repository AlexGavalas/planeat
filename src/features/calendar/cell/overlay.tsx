import {
    ActionIcon,
    type ActionIconProps,
    Center,
    Overlay,
    SimpleGrid,
    Tooltip,
} from '@mantine/core';
import { useModals } from '@mantine/modals';
import { EditPencil, Notes, ThreeStars } from 'iconoir-react';
import { useTranslation } from 'next-i18next';
import { useCallback } from 'react';

import { CopyButton } from '~components/copy-button';
import { MealModal } from '~features/modals/meal';
import { MealNoteModal } from '~features/modals/meal-note';
import { MealRatingModal } from '~features/modals/meal-rating';
import { type EditedMeal, type Meal } from '~types/meal';

type CellOverlayProps = {
    onDelete: () => Promise<void> | void;
    onSave: (value: Partial<Meal>) => Promise<void> | void;
    meal?: Meal | EditedMeal;
};

const isSavedMeal = (meal?: Meal | EditedMeal): meal is Meal => {
    return meal?.id !== undefined;
};

const isFilledMeal = (meal?: Meal | EditedMeal): meal is Meal => {
    return Boolean(meal?.meal);
};

const commonButtonProps = {
    size: 'lg',
    variant: 'light',
} satisfies ActionIconProps;

export const CellOverlay = ({ onDelete, onSave, meal }: CellOverlayProps) => {
    const { t } = useTranslation();
    const modals = useModals();

    const isMealSaved = isSavedMeal(meal);
    const isMealFilled = isFilledMeal(meal);

    const handleMealSave = useCallback(
        async (value: string) => {
            await onSave({ ...meal, meal: value });
        },
        [onSave, meal],
    );

    const handleMealNoteSave = useCallback(
        async (note: string) => {
            await onSave({ ...meal, note });
        },
        [onSave, meal],
    );

    const handleMealNoteDelete = useCallback(async () => {
        await onSave({ ...meal, note: null });
    }, [onSave, meal]);

    const handleMealRatingSave = useCallback(
        async (rating: number) => {
            await onSave({ ...meal, rating });
        },
        [onSave, meal],
    );

    const handleMealRatingDelete = useCallback(async () => {
        await onSave({ ...meal, rating: null });
    }, [onSave, meal]);

    const handleEditClick = useCallback(() => {
        modals.openModal({
            centered: true,

            children: (
                <MealModal
                    initialMeal={meal?.meal ?? ''}
                    onDelete={onDelete}
                    onSave={handleMealSave}
                />
            ),
            title: t('edit_meal'),
        });
    }, [onDelete, handleMealSave, meal?.meal, modals, t]);

    const handleNoteClick = useCallback(() => {
        if (!isMealSaved) {
            return;
        }

        modals.openModal({
            centered: true,
            children: (
                <MealNoteModal
                    meal={meal}
                    onDelete={handleMealNoteDelete}
                    onSave={handleMealNoteSave}
                />
            ),
            title: t('notes'),
        });
    }, [
        handleMealNoteDelete,
        handleMealNoteSave,
        isMealSaved,
        meal,
        modals,
        t,
    ]);

    const handleRateClick = useCallback(() => {
        if (!isMealSaved) {
            return;
        }

        modals.openModal({
            centered: true,
            children: (
                <MealRatingModal
                    meal={meal}
                    onDelete={handleMealRatingDelete}
                    onSave={handleMealRatingSave}
                />
            ),
            title: t('modals.meal_rate.title'),
        });
    }, [
        handleMealRatingDelete,
        handleMealRatingSave,
        isMealSaved,
        meal,
        modals,
        t,
    ]);

    const shouldShowColumnLayout = isMealSaved || isMealFilled;

    return (
        <Overlay style={{ background: 'rgba(255, 255, 255, 0.5)' }}>
            <Center style={{ height: '100%' }}>
                <SimpleGrid
                    cols={shouldShowColumnLayout ? 2 : 1}
                    spacing={4}
                    verticalSpacing={4}
                >
                    <Tooltip
                        withArrow
                        label={t('generic.actions.edit')}
                        position={shouldShowColumnLayout ? 'left' : 'top'}
                    >
                        <ActionIcon
                            {...commonButtonProps}
                            onClick={handleEditClick}
                        >
                            <EditPencil />
                        </ActionIcon>
                    </Tooltip>
                    {isMealSaved && (
                        <>
                            <Tooltip
                                withArrow
                                label={t('tooltip.edit_note')}
                                position="right"
                            >
                                <ActionIcon
                                    {...commonButtonProps}
                                    onClick={handleNoteClick}
                                >
                                    <Notes />
                                </ActionIcon>
                            </Tooltip>
                            <Tooltip
                                withArrow
                                label={t('tooltip.rate')}
                                position="left"
                            >
                                <ActionIcon
                                    {...commonButtonProps}
                                    onClick={handleRateClick}
                                >
                                    <ThreeStars />
                                </ActionIcon>
                            </Tooltip>
                        </>
                    )}
                    {isMealFilled && (
                        <CopyButton tooltipPosition="right" value={meal.meal} />
                    )}
                </SimpleGrid>
            </Center>
        </Overlay>
    );
};
