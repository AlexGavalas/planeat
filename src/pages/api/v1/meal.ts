import { createPagesServerClient } from '@supabase/auth-helpers-nextjs';
import type { NextApiHandler } from 'next';
import invariant from 'tiny-invariant';
import { ZodError } from 'zod';

import { createMeals, deleteMeals, fetchMeals, updateMeals } from '~api/meal';
import { getServerSession } from '~api/session';
import { fetchUser } from '~api/user';
import { patchRequestSchema } from '~schemas/meal';
import { assertSession, assertUserEmail } from '~util/session';

const handler: NextApiHandler = async (req, res) => {
    try {
        const session = await getServerSession({ req, res });

        assertSession(session);
        assertUserEmail(session.user?.email);

        const supabase = createPagesServerClient({ req, res });

        const user = await fetchUser({ supabase, email: session.user.email });

        invariant(user, 'User must exist');

        if (req.method === 'GET') {
            const endDate = String(req.query.endDate);
            const startDate = String(req.query.startDate);

            const { data } = await fetchMeals({
                endDate,
                startDate,
                supabase,
                userId: user.id,
            });

            res.json({ data });
        } else if (req.method === 'PATCH') {
            const { deletedIds, editedMeals, newMeals } =
                patchRequestSchema.parse(req.body);

            const { error: deleteError } = await deleteMeals({
                deletedIds,
                supabase,
                userId: user.id,
            });

            const { error: updateError } = await updateMeals({
                editedMeals,
                supabase,
                userId: user.id,
            });

            const { error: createError } = await createMeals({
                newMeals,
                supabase,
                userId: user.id,
            });

            const error = deleteError ?? updateError ?? createError;

            if (error) {
                throw new Error(error.message, { cause: error });
            }

            res.status(200).json({ message: 'OK' });
        } else {
            res.status(405).json({ message: 'Method Not Allowed' });
        }
    } catch (e) {
        console.error(e);

        if (e instanceof ZodError) {
            res.status(400).json({ message: 'Bad Request' });
        } else {
            res.status(500).json({ message: 'Internal Server Error' });
        }
    }
};

export default handler;
