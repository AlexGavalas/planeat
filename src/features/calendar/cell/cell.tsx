import { Box, Center, Text } from '@mantine/core';
import { useHover } from '@mantine/hooks';
import { MultiplePages } from 'iconoir-react';

import { useProfile } from '~hooks/use-profile';
import { useMeals } from '~store/hooks';
import { type EditedMeal, type Meal } from '~types/meal';

import styles from './cell.module.css';
import { CellOverlay } from './overlay';

export type CellProps = {
    id: string;
    meal?: Meal | EditedMeal;
    timestamp: Date;
    isEdited: boolean;
    isRow: boolean;
};

export const Cell = ({ id, meal, timestamp, isEdited, isRow }: CellProps) => {
    const { profile } = useProfile();
    const { hovered, ref } = useHover();

    const { deleteEntryCell, deleteEntryRow, saveEntryCell, saveEntryRow } =
        useMeals();

    const handleSave = async (newMeal: Partial<Meal>) => {
        if (!profile || !newMeal.meal) {
            return;
        }

        if (isRow) {
            saveEntryRow({
                sectionKey: id,
                userId: profile.id,
                value: newMeal.meal,
                note: newMeal.note,
                rating: newMeal.rating,
            });
        } else {
            saveEntryCell({
                meal,
                sectionKey: id,
                timestamp,
                userId: profile.id,
                value: newMeal.meal,
                note: newMeal.note,
                rating: newMeal.rating,
            });
        }
    };

    const handleDelete = async () => {
        if (!meal) {
            return;
        }

        if (isRow) {
            deleteEntryRow(id);
        } else {
            deleteEntryCell({ meal });
        }
    };

    const hasNote = !!meal?.note;

    return (
        <Box
            ref={ref}
            style={{
                position: 'relative',
                ...(isRow && { gridColumn: 'span 7' }),
            }}
        >
            {hovered && (
                <CellOverlay
                    handleDelete={handleDelete}
                    handleSave={handleSave}
                    meal={meal}
                />
            )}
            {hasNote && (
                <Box className={styles.infoContainer}>
                    <MultiplePages />
                </Box>
            )}
            <Box
                className={styles.cell}
                style={{
                    ...(isEdited && { border: '2px solid orange' }),
                }}
            >
                <Center
                    style={{
                        ...(hovered && { opacity: 0.15 }),
                    }}
                >
                    <Text p={5} align="center">
                        {meal?.meal || 'N/A '}
                    </Text>
                </Center>
            </Box>
        </Box>
    );
};
