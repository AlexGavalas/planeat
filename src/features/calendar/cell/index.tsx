import { Box, Center, Text } from '@mantine/core';
import { useHover } from '@mantine/hooks';
import { MultiplePages } from 'iconoir-react';

import { useProfile } from '~hooks/use-profile';
import { useMeals } from '~store/hooks';

import styles from './cell.module.css';
import { CellOverlay } from './overlay';

export const Cell = ({ id, meal, timestamp, isEdited, isRow }: CellProps) => {
    const { profile: user } = useProfile();
    const { hovered, ref } = useHover();

    const { deleteEntryCell, deleteEntryRow, saveEntryCell, saveEntryRow } =
        useMeals();

    const handleSave = async (newMeal: Partial<Meal>) => {
        if (!user || !newMeal.meal) {
            return;
        }

        if (isRow) {
            saveEntryRow({
                sectionKey: id,
                userId: user.id,
                value: newMeal.meal,
                note: newMeal.note,
            });
        } else {
            saveEntryCell({
                meal,
                sectionKey: id,
                timestamp,
                userId: user.id,
                value: newMeal.meal,
                note: newMeal.note,
            });
        }
    };

    const handleDelete = async () => {
        if (!meal) {
            return;
        }

        if (isRow) {
            deleteEntryRow({ id, meal });
        } else {
            deleteEntryCell({ id, meal });
        }
    };

    const hasNote = !!meal?.note;

    return (
        <Box
            ref={ref}
            style={{
                position: 'relative',
                ...(isRow && { gridColumn: 'span 7' }),
            }}
        >
            {hovered && (
                <CellOverlay
                    handleDelete={handleDelete}
                    handleSave={handleSave}
                    meal={meal}
                />
            )}
            {hasNote && (
                <Box className={styles.infoContainer}>
                    <MultiplePages />
                </Box>
            )}
            <Box
                className={styles.cell}
                style={{
                    ...(isEdited && { border: '2px solid orange' }),
                }}
            >
                <Center
                    style={{
                        ...(hovered && { opacity: 0.15 }),
                    }}
                >
                    <Text p={5} align="center">
                        {meal?.meal || 'N/A '}
                    </Text>
                </Center>
            </Box>
        </Box>
    );
};
