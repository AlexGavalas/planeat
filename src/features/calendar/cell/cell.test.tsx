import { renderWithUser } from '~test/utils';
import { type Meal } from '~types/meal';

import { Cell, type CellProps } from './cell';

jest.mock('~hooks/use-profile', () => ({
    useProfile: jest.fn().mockReturnValue({}),
}));

jest.mock('~store/hooks', () => ({
    useMeals: jest.fn().mockReturnValue({
        deleteEntryCell: jest.fn(),
        deleteEntryRow: jest.fn(),
        saveEntryCell: jest.fn(),
        saveEntryRow: jest.fn(),
    }),
}));

describe('<Cell />', () => {
    const props: CellProps = {
        id: 'id',
        isEdited: false,
        isRow: false,
        timestamp: new Date(),
    };

    it('renders', () => {
        const { container } = renderWithUser(<Cell {...props} />);

        expect(container).toMatchSnapshot();
    });

    describe('when the user has a meal', () => {
        const meal: Meal = {
            id: 'meal-id',
            meal: 'test meal',
            day: '2021-01-01',
            section_key: 'section-key',
            user_id: 1,
            note: null,
        };

        const propsWithMeal: CellProps = {
            ...props,
            meal,
        };

        it('renders', () => {
            const { container } = renderWithUser(<Cell {...propsWithMeal} />);

            expect(container).toMatchSnapshot();
        });

        describe('when the meal has a note', () => {
            const propsWithMealNote: CellProps = {
                ...props,
                meal: {
                    ...meal,
                    note: 'test note',
                },
            };

            it('renders', () => {
                const { container } = renderWithUser(
                    <Cell {...propsWithMealNote} />,
                );

                expect(container).toMatchSnapshot();
            });
        });
    });
});
