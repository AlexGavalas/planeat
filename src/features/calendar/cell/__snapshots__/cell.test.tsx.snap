// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> renders 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="position: relative;"
  >
    <div
      class="cell"
    >
      <div
        class="m-4451eb3a mantine-Center-root"
      >
        <p
          class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
          style="text-align: center;"
        >
          N/A
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cell /> when the user has a meal renders 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="position: relative;"
  >
    <div
      class="cell"
    >
      <div
        class="m-4451eb3a mantine-Center-root"
      >
        <p
          class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
          style="text-align: center;"
        >
          test meal
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cell /> when the user has a meal when the meal has a note renders 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="position: relative;"
  >
    <div
      class="infoContainer"
    >
      <svg
        color="currentColor"
        fill="none"
        height="1.5em"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 18h7M7 14h1M7 10h3M7 2h9.5L21 6.5V19"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <path
          d="M3 20.5v-14A1.5 1.5 0 014.5 5h9.752a.6.6 0 01.424.176l3.148 3.148A.6.6 0 0118 8.75V20.5a1.5 1.5 0 01-1.5 1.5h-12A1.5 1.5 0 013 20.5z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <path
          d="M14 5v3.4a.6.6 0 00.6.6H18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <div
      class="cell"
    >
      <div
        class="m-4451eb3a mantine-Center-root"
      >
        <p
          class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
          style="text-align: center;"
        >
          test meal
        </p>
      </div>
    </div>
  </div>
</div>
`;
